oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/LogoutHandler.java

```diff
@@ -92,135 +92,63 @@ public APIGatewayProxyResponseEvent handleRequest(
                 "oidc-api::" + getClass().getSimpleName(), () -> logoutRequestHandler(input));
     }
 
-    public APIGatewayProxyResponseEvent logoutRequestHandler(APIGatewayProxyRequestEvent input) {
+    public APIGatewayProxyResponseEvent logoutRequestHandler(APIGatewayProxyRequestEvent input)
+            throws ParseException {
         LOG.info("Logout request received");
-        Optional<String> state;
-        if (input.getQueryStringParameters() == null
-                || input.getQueryStringParameters().isEmpty()) {
-            LOG.info("No query string parameters in request");
-            state = Optional.empty();
-        } else {
-            state = Optional.ofNullable(input.getQueryStringParameters().get("state"));
-        }
         Optional<Session> sessionFromSessionCookie =
                 segmentedFunctionCall(
                         "getSessionFromSessionCookie",
                         () -> sessionService.getSessionFromSessionCookie(input.getHeaders()));
-        if (sessionFromSessionCookie.isPresent()) {
-            return segmentedFunctionCall(
-                    "processLogoutRequest",
-                    () -> processLogoutRequest(sessionFromSessionCookie.get(), input, state));
+        Map<String, String> queryStringParameters = input.getQueryStringParameters();
+        Optional<String> state;
+        if (queryStringParameters == null || queryStringParameters.isEmpty()) {
+            LOG.info("Returning default logout as no input parameters");
+            return defaultLogoutWithSessionCheck(sessionFromSessionCookie, input, Optional.empty());
```

Comment:
Good point - see what you mean with the implication. Can separate out as suggested if it makes it clearer.

Reply:
The logout is tied to the `session check` where it use the `sessionId` if the session exists in the logout. Therefore it feels like it makes sense for it to be in the same method - maybe a better method name ? Happy to be told otherwise.

Reply:
Similar point as before - why does it need to specify it is destroying the session if exists? Is that not a standard part of logging out?


Additionally there is `logoutWithSession` that also destroys the session - feels a bit inconsistent

Reply:
There are default logouts without destroying the session. Can understand it feels inconsistent as "checks" that take place are now before the session check. However to keep the same logouts before , I've had to extract the checking of session and destruction in different places.

Reply:
Previous logic indicates that sessions should not be destroyed in error logouts unless it is erroring due to:
1. The clientID not being present in the client registry
2. The client registry does not have the post redirect uri